packages/orchestration/src/types.ts

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable no-use-before-define -- fine for types */
```

Comment:
why a type.ts file rather than type.d.ts?

Reply:
because a `.d.ts` file has two downsides,
1. requires an empty `.js` file next to it
2. prevents type checking of the file contents (because tsc trusts `.d.ts` as its own output)

I'll try to document this justification

Reply:
https://github.com/Agoric/agoric-sdk/pull/9364/